### DISCUSSION TITLE
trying to read data from the DFRobot TEL0168 BLE sensor beacon

### DISCUSSION THREAD

**Discussion by jfranklyn** [hbldh/bleak#1569] [Q&A]
Hello all,
I'm using a DFRobot TEL0168 BLE sensor beacon with your code, trying to read the data from a soil moisture sensor attached to the device. The sample code has a required argument of UUID. This device is configured as a custom device, not ibeacon or eddystone, so it does not have a UUID that I can generate or configure. I can use some of the sample code to list the device. Can I modify the code so that I can read from this device? I've included the wiki for this device below. Any help would be greatly appreciated. Thanks

https://wiki.dfrobot.com/_SKU_TEL0168_Fermion_BLE_Sensor_Beacon

**Comment by dlech** [comment]
It sounds like it just does advertising data and doesn't actually connect. So you would just need something like the detection callback or scan iterator example programs [here](https://github.com/hbldh/bleak/tree/develop/examples).
